runtime/src/system_instruction_processor.rs

```diff
@@ -179,6 +179,13 @@ fn transfer_verified(
         ic_msg!(invoke_context, "Transfer: `from` must not carry data");
         return Err(InstructionError::InvalidArgument);
     }
+
+    if invoke_context.is_feature_active(&feature_set::system_transfer_zero_check::id())
+        && lamports == 0
+    {
+        return Ok(());
+    }
+
```

Comment:
this looks unnecessary. The lamports check below will be bypassed when `lamports == 0`

Reply:
Yeah it looks a bit silly, but it avoids unnecessary calls to `try_account_ref_mut()` later on.  If there's no possible issues with doing those, I'm happy to get rid of this

Reply:
yeah no possible issues with that, let's get rid of it